### ISSUE TITLE
Proposal for additional package that includes configurations for running Grafana Agent in Flow mode for debian-based systems.

### ISSUE THREAD

**Issue by robigan** [grafana/agent#3303]
Hello world!
I would like to propose a package for **debian-based** systems (that can be tweaked for other OSes such as rpm) that I made earlier today that could be useful towards the adoption of Grafana agent's flow operational mode. This is an additional/alternative package that includes various configurations and systemd service unit for running Grafana agent in flow mode either side-by-side to standard Grafana agent or to allow switching solely to Grafana agent flow. 

This package that I made (which resides in this [repo I made](https://github.com/robigan/grafana-agent-flow)) I think is the best way to make it as easy as possible to adopt Grafana agent flow without much additional maintenance as it includes the systemd service unit for running Grafana agent flow which just makes it easier to switch over to flow without having to mess with systemd service unit files and the associated `EnvironmentFile` required by the systemd service unit. Also, given that the approach I am suggesting simply creates an additional service unit and configuration directory, it makes adopting flow an incremental process without having to ship two separate packages that both include the full grafana-agent binary. This is less significant, but I've seen various developers ship separate packages with essentially the same binary when phasing one codebase out and another in.

Could this be useful? I made this because I needed an easy way to run grafana-agent in flow mode which I could ship in a package to my various machines which had Grafana agent installed which I wished to switch over to flow mode.

**Quick links**
Repo: [https://github.com/robigan/grafana-agent-flow](https://github.com/robigan/grafana-agent-flow)

**Comment by rfratto** [comment]
👋 I think this makes sense, though we'll need to think about how we want to do it. We currently use FPM for generating .debs and .rpms, so we'll probably want to continue using that for Flow-mode specific packages. 

I also think this should apply to non-Linux systems too, with us producing a Flow-specific Homebrew formula for macOS and probably a separate Flow-focused installer for Windows.

**Comment by robigan** [comment]
> 👋 I think this makes sense, though we'll need to think about how we want to do it. We currently use FPM for generating .debs and .rpms, so we'll probably want to continue using that for Flow-mode specific packages. 
> 
> 
> 
> I also think this should apply to non-Linux systems too, with us producing a Flow-specific Homebrew formula for macOS and probably a separate Flow-focused installer for Windows. 

Sorry for my late response, I've been busy. I see, I can try to 'configure' fpm to make flow specific packages but I have no experience with it so I'll have to see how far my skills go. Homebrew I should know well enough to make a package for that too. As for Windows, I haven't touched it in years so Idek how to make an installer.

**Comment by rfratto** [comment]
I'm planning on getting things together for this over the next few weeks. I'll give you credit in the changelog if you're interested :)

**Comment by robigan** [comment]
> I'm planning on getting things together for this over the next few weeks. I'll give you credit in the changelog if you're interested :)

Yeah why not if you wish. I hardly check github notifs so sorry for late response
